bpo-39573: Add Py_IS_TYPE macro (#18488)
python/cpython: Doc/c-api/structures.rst

      (((PyObject*)(o))->ob_type)

.. c:function:: int Py_IS_TYPE(PyObject *o, PyTypeObject *type)

>    Return true if the type of object *o* is the *type*.

Reviewer: ```suggestion
   Return non-zero if the object *o* type is *type*. Return zero otherwise.
```

>    Equivalent to: Py_TYPE(o) == type.

Reviewer: ```suggestion
   Equivalent to: ``Py_TYPE(o) == type``.
```

   .. versionadded:: 3.9

> 

Reviewer: Ooops, I noticed a typo in one of my previous commit. Would you mind to take this PR as an opportunity to fix it?

https://docs.python.org/dev/c-api/structures.html#c.Py_SET_SIZE

"Set the object o size of size." => "Set ... to size."

Author: Oh sure :)

Reviewer: Wait, @brandtbucher wrote PR #18496. I will ask him to fix the typo there. You can ignore this special request ;-)

Author: Got it ;-)

Author: ## Current
<img width="760" alt="스크린샷 2020-02-13 오후 11 23 37" src="https://user-images.githubusercontent.com/5110323/74444397-29c5cf80-4eb8-11ea-9421-715c3a677aea.png">

## Suggestion
<img width="826" alt="스크린샷 2020-02-13 오후 11 15 15" src="https://user-images.githubusercontent.com/5110323/74444400-2b8f9300-4eb8-11ea-89c9-9ffcf3615213.png">

Looks like not work as we want ;)

      (((PyObject*)(o))->ob_type)

> .. c:macro:: Py_IS_TYPE(o, type)

Reviewer: ```suggestion
 .. c:function:: int _Py_IS_TYPE(PyObject *ob, PyTypeObject *type)
```

>    This macro is used to check whether the type of object *o* is the *type*.

Reviewer: I suggest to remove "This macro ..." and only describe the behavior. See Py_SET_REFCNT() for example.

I also suggest to remove the "It expands to ..." part, since it's no longer a macro and I don't think that it's worth it to decribe the implementation.

Or maybe add something like: ``Equivalent to: Py_TYPE(o) == type.``

   It expands to::

>       (((PyObject*)(o))->ob_type) == (type))

Reviewer: I suggest:

Check whether the type of object *o* is equal to *type*. Equivalent to `Py_TYPE(o) == type`.

I dislike documentating the exact implementation.
